### PR TITLE
prefetch-npm-deps: add support for NIX_NPM_TOKENS env var

### PR DESCRIPTION
**Pull request by lilyinstarlight**
## Description of changes

This was tested and confirmed working for @countoren in `#nodejs:nixos.org` on Matrix

Basically it allows using impure NPM registry tokens for private registries (e.g. fontawesome)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/node/fetch-npm-deps/default.nix [hunk 1]
```diff
@@ -165,7 +165,7 @@
 
       dontInstall = true;
 
-      impureEnvVars = lib.fetchers.proxyImpureEnvVars;
+      impureEnvVars = lib.fetchers.proxyImpureEnvVars ++ [ "NIX_NPM_TOKENS" ];
```
Comment (Mindavi) [inline pkgs/build-support/node/fetch-npm-deps/default.nix:168]: It might make sense to document the format here as well.

Comment (lilyinstarlight) [reply pkgs/build-support/node/fetch-npm-deps/default.nix:168]: Yeah, I agree that's a good idea. I'll do that when I'm next at my computer. Thank you!

I _should_ also add it to the tooling docs, so maybe I'll do that too

Comment (lilyinstarlight) [reply pkgs/build-support/node/fetch-npm-deps/default.nix:170]: Done! Apparently there still is no documentation on `fetchNpmDeps` in general so, uh, I didn't include an update to the tooling docs that apparently don't exist 😅

**Comment by lilyinstarlight** [comment]
Backport probably needs #242267

I'll assess more later if no one else does first (at $dayjob on mobile phone right now)

**Pull request merged**